jaxrs-spec/src/main/asciidoc/chapters/resources/_resource-classes.adoc

```diff
@@ -40,10 +40,10 @@ parameter values. Note that a zero argument constructor is permissible
 under this rule.
 
 A public constructor MAY include parameters annotated with one of the
-following: `@Context`, `@HeaderParam`, `@CookieParam`, `@MatrixParam`,
-`@QueryParam` or `@PathParam`. However, depending on the resource class
-lifecycle and concurrency, per-request information may not make sense in
-a constructor. If more than one public constructor is suitable then an
+following: `@HeaderParam`, `@CookieParam`, `@MatrixParam`, `@QueryParam` or
+`@PathParam`. However, depending on the resource class lifecycle and
+concurrency, per-request information may not make sense in a
+constructor. If more than one public constructor is suitable then an
 implementation MUST use the one with the most parameters. Choosing
```

Comment:
I don't know if we can inject these in constructors anymore. We can in resource methods since we should be in control of the invocation. Perhaps we could do this in constructors too, but would need some CDI expert to help us here.

Reply:
I've asked Emily Jiang and a few others to give an opinion.

Reply:
It seems to be similar to `@Valid` for instance. It could be easier for the customers to transition to 4.0 allowing it, but I am not sure what CDI shenanigans it would mean to support these annotations right now.

Reply:
I'll need to go back and read the CDI issues, but my assumption or thought, was that these are CDI qualifiers?